src/ODEs/ODETools/EXRungeKutta.jl

```diff
@@ -0,0 +1,148 @@
+"""
+Explicit Runge-Kutta ODE solver
+"""
+struct EXRungeKutta <: ODESolver
+  nls::NonlinearSolver
+  dt::Float64
+  tableau::EXButcherTableau
+  function EXRungeKutta(nls::NonlinearSolver, dt, type::Symbol)
+    bt = EXButcherTableau(type)
+    new(nls, dt, bt)
+  end
+end
+
+"""
+solve_step!(uf,odesol,op,u0,t0,cache)
+"""
+function solve_step!(uf::AbstractVector,
+  solver::EXRungeKutta,
+  op::ODEOperator,
+  u0::AbstractVector,
+  t0::Real,
+  cache)
+
+  # Unpack variables
+  dt = solver.dt
+  s = solver.tableau.s
+  a = solver.tableau.a
+  b = solver.tableau.b
+  c = solver.tableau.c
+  d = solver.tableau.d
+
+  # Create cache if not there
+  if cache === nothing
+    ode_cache = allocate_cache(op)
+    vi = similar(u0)
+    ki = [similar(u0) for i in 1:s]
+    M = allocate_jacobian(op,t0,uf,ode_cache)
+    get_mass_matrix!(M,op,t0,uf,ode_cache)
+    nl_cache = nothing
+  else
+    ode_cache, vi, ki, M, nl_cache = cache
+  end
+
+  nlop = EXRungeKuttaStageNonlinearOperator(op,t0,dt,u0,ode_cache,vi,ki,0,a,M)
+
+  for i in 1:s
+
+    # solve at stage i
+    ti = t0 + c[i]*dt
+    ode_cache = update_cache!(ode_cache,op,ti)
+    update!(nlop,ti,ki[i],i)
+    nl_cache = solve!(uf,solver.nls,nlop,nl_cache)
```

Comment:
this is the LinearSolver cache, no nonlinear

Reply:
changed to l_cache